main.rb

```diff
@@ -0,0 +1,149 @@
+require './student'
+require './teacher'
+require './book'
+require './rental'
+
+def create_student(age, name)
+  print('Has Parent Permission [y/n]:')
+  permission = gets.chomp
+  if permission == 'y'
+    Student.new(age, name)
+  else
+    Student.new(age, name, parent_permission: false)
+  end
+end
+
+def create_teacher(age, name)
+  print('Specialization:')
+  specialization = gets.chomp
+  Teacher.new(age, specialization, name)
+end
+
+def create_person
+  print('Do you want to create a student(1) or a teacher (2) [Input the number]:')
+  input = gets.chomp
+  print('Age:')
+  age = gets.chomp
+  print('Name:')
+  name = gets.chomp
+  case input
+  when '1'
+    @persons << create_student(age, name)
+  when '2'
+    @persons << create_teacher(age, name)
+  else
+    puts('Wrong Input')
+    create_person
+  end
+  puts('Person Created succefully')
+end
+
+def create_book
+  print('Title:')
+  title = gets.chomp
+  print('Author:')
+  author = gets.chomp
+  @books << Book.new(title, author)
+  puts('Book Created succefully')
+end
+
+def list_books
+  i = 0
+  @books.each do |book|
+    puts("#{i}) title: #{book.title}, author: #{book.author}")
+    i += 1
+  end
+end
+
+def list_persons
+  i = 0
+  @persons.each do |person|
+    if person.is_a?(Teacher)
+      print '[Teacher] '
+    else
+      print '[Student] '
+    end
+    puts("#{i}) Name: #{person.name}, ID: #{person.id}, Age: #{person.age}")
+    i += 1
+  end
+end
+
+def person_object(id)
+  @persons.each do |person|
+    return person if person.id == id
+  end
+  nil
+end
+
+def print_rentals
+  print('Person ID: ')
+  id = gets.chomp.to_i
+  person = person_object(id)
+  if person.nil?
+    puts('Wrong Id')
+  elsif person.rentals == []
+    puts('No rentals for this person')
+  else
+    person.rentals.each do |rent|
+      puts("Date: #{rent.date}, Book: #{rent.book.title} by #{rent.book.title}")
+    end
+  end
+end
+
+def create_new_rental
+  puts('Select a book from following list by number:')
+  list_books
+  book_index = gets.chomp.to_i
+  puts('Select a person from following list by number:')
+  list_persons
+  person_index = gets.chomp.to_i
+  print('Date: ')
+  date = gets.chomp
+  Rental.new(date, @persons[person_index], @books[book_index])
+  puts('Rental created succefully')
+end
```

Comment:
Look at the image below

- [ ] You still have not resolved the issue of adding a new rental as highlighted by the previous reviewer. When I try to create a rental I get the **`NoMethodError`** error. Now, this is because in your `rental.rb` file you are calling `person.rentals` but in your `person.rb` class you initialized `rental`, instead of **`rentals`**. Kindly change **`rental`** to **`rentals`**

![namutenda-more-err](https://user-images.githubusercontent.com/36542986/144590770-f67788aa-58b8-4b16-aff7-ea46622c3822.jpg)

Reply:
Fixed